AV-101592 Update index.md and upgrade guide (#330)
couchbasecloud/terraform-provider-couchbase-capella: docs/index.md

  Couchbase Capella uses an ordered hierarchy to help you keep all of your data organized and securely accessible.

>   The entity at the top of the hierarchy is called an organization.

Reviewer: ```suggestion
  The hierarchy starts with an organization.
```

Author: I'm gonna keep it as it is as the previous one is more clear.

>   Everything you do in Capella  --  whether it's creating a cluster or managing billing  --  happens within the scope of an [organization](https://docs.couchbase.com/cloud/organizations/organizations.html).

Reviewer: ```suggestion
  Everything you do in Capella, from creating a cluster to managing your billing, happens inside an [organization](https://docs.couchbase.com/cloud/organizations/organizations.html).
```

>   Database credentials are separate from organization roles and project roles.

Reviewer: Should we be saying "clusters" instead of "databases"? I know in the Management REST API reference doc, we have this issue as well. So this may be a fix later situation, but I wanted to flag it in case. :)

Author: Let's keep as it is for now, we can discuss and pick it up later.

>   A [database credential](https://docs.couchbase.com/cloud/clusters/manage-database-users.html#about-database-credentials) is specific to a database and consists of a database access name, secret, and a set of bucket and scope access levels.

Reviewer: ```suggestion
  A [database credential](https://docs.couchbase.com/cloud/clusters/manage-database-users.html#about-database-credentials), also known as a cluster access credential, is specific to an operational cluster and consists of a cluster access name, secret, and a set of bucket and scope access levels.
```

Reviewer: This suggestion may be tricky as I know in the management API, this API is still written as "Database credential", but in our documentation (which is referenced), it's labeled as "Cluster access credential". I tried to find a way to make this clear to the user and include both options we use.

Author: IMO we should keep this as Database Credential, we need to change it throughout the all the docs. If Product team and confirm and suggests the same, this can be taken as an enhancement

Reviewer: Sounds like a plan. 👍

>   It's required for applications to remotely authenticate on a database and access bucket data.

Reviewer: ```suggestion
  It's required for applications to remotely authenticate on an operational cluster and access bucket data.
```

Author: I'll keep it as database as it is addressed the same in the mgmt api and everywhere else. IMO we can update it in all places as an enhancement, there should be uniformity everywhere or the user will be confused.

* [Manage Flush Bucket](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/tree/main/examples/flush_bucket):

  Flush a bucket to have the system delete all its data at the earliest opportunity available.

* [Manage GSI](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/tree/main/examples/gsi):

  Manage secondary indexes on a Capella operational cluster. It's recommended to use deferred index builds, especially for larger indexes.

* [Create & Manage Free Tier Clusters](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/tree/main/examples/free_tier_cluster):

>   Create and manage a free tier cluster in Capella. Free tier clusters provide a cost-effective way to get started with Couchbase Capella. This is a 1 node cluster that only runs data, query, index and search services. There can be at most 1 free tier cluster per tenant.

Reviewer: ```suggestion
  Create and manage a free tier operational cluster in Capella. Free tier operational clusters provide a cost-effective way to get started with Couchbase Capella. This is a Single Node cluster that runs only the Data, Query, Index, and Search services. Only 1 free tier operational cluster is available per organization
```

* [Create & Manage Free Tier Buckets](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/tree/main/examples/free_tier_bucket):

>   Create and manage buckets within a free tier cluster. Free tier buckets have specific memory allocation limits. This is a Couchbase bucket where only the name a memory quota is configurable. Other bucket properties use default values.

Reviewer: ```suggestion
  Create and manage buckets within a free tier operational cluster. Free tier buckets have specific memory allocation limits. This is a Couchbase bucket where only the name and memory quota is configurable. Other bucket properties use default values.
```

* [Create & Manage Free Tier App Services](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/tree/main/examples/free_tier_appservice):

>   Set up and manage App Services for your free tier cluster. This is a 1 node cluster which can only be linked to a free tier cluster. It can only be turned off/on when the linked free tier cluster is turned off/on.

Reviewer: ```suggestion
  Set up and manage App Services for your free tier operational cluster. This is a Single Node App Service that can only be linked to a free tier operational cluster. The App Service can only be turned off and on when the linked free tier operational cluster is turned off and on.
```

* [Manage Free Tier Cluster On/Off](https://github.com/couchbasecloud/terraform-provider-couchbase-capella/tree/main/examples/free_tier_cluster_on_off):

>   Turn clusters on or off on demand and control when your free tier cluster is running to optimize resource usage. Turning the free tier cluster on or off will also turn on or off the linked app services, if any. 

Reviewer: ```suggestion
  Turn you free tier operational cluster on or off on demand to optimize resource usage. Turning the free tier cluster on or off will also turn on or off any linked App Services. 
```
